src/utils/timeout.ts

```diff
@@ -24,12 +24,12 @@ export async function valueOnTimeout<T>(timeoutMs: number, timeoutValue: T, acti
 /**
  * Returns the result of awaiting a specified action. Rejects if the action throws or if the time-out occurs.
  */
-export async function rejectOnTimeout<T>(timeoutMs: number, action: () => Promise<T> | T) {
+export async function rejectOnTimeout<T>(timeoutMs: number, action: () => Promise<T> | T, callerTimeOutError?: string) {
```

Comment:
callerTimeOutMessage

It would actually be useful in the future to allow an Error object, but for now please be more specific about what it is, thanks.